sympy/combinatorics/homomorphisms.py

```diff
@@ -437,6 +437,10 @@ def group_isomorphism(G, H):
     First, the set of generators are mapped with the elements of `H` and we check if the mapping induces an isomorphism.
 
     '''
+    # Two infinte FpGroups with the same generators are isomorphic when the relators are same but are ordered differently.
+    if isinstance(G, FpGroup) and isinstance(H, FpGroup):
+        if G.generators == H.generators and (G.relators).sort() == (H.relators).sort():
+            return homomorphism(G, H, G.generators, H.generators)
```

Comment:
It could be worth adding a case that uses `simplify_presentation` for infinite groups because you could theoretically end up in this situation after the presentation is simplified. For example,
```
F, a, b = free_group("a b")
E, c, d, e = free_group("c d e")
G = FpGroup(F, [a**2])
H = FpGroup(E, [b**2, c])
```
`G` and `H` are isomorphic because `c` is just a redundant element. `simplify_presentation(H)` should return a group on generators `x_0, x_1` and with relators `x_1**2`, or something like that - that's clearly the same as `G`.

We would also like this behavior:
```
F, a, b = free_group("a b")
G = FpGroup(F, [a**2])
H = FpGroup(F, [b**2])
```
which doesn't happen at the moment.

Reply:
> >>> H = FpGroup(E, [b**2, c])

Shouldn't the relators be of the same free group?

Reply:
> Shouldn't the relators be of the same free group?
No because these groups are isomorphic, and that's the only thing that should matter for this method.